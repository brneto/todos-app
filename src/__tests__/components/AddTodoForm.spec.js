import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import AddTodoForm from '../../client/components/AddTodoForm';

describe('components/AddTodo', () => {
  it('should call handle methods', () => {
    const handleChange = jest.fn();
    const handleKeyDown = jest.fn();

    // render the todo form with a fake key down handler
    const formProps = {
      value: '',
      onChange: handleChange,
      onKeyDown: handleKeyDown,
    };
    const { getByPlaceholderText, container } = render(
      <AddTodoForm input={formProps} />
    );

    // get the todo input
    const todoInput = getByPlaceholderText('What needs to be done?');
    // Fires onChange event
    fireEvent.change(todoInput, { target: { value: 'change' } });
    // Fires onKeyDown event
    fireEvent.keyDown(todoInput);

    // assert that our fake submit handler was called once
    expect(handleChange).toHaveBeenCalledTimes(1);
    // assert that our fake submit was called with the right stuff
    expect(handleKeyDown).toHaveBeenCalledTimes(1);
    // assert if equal to snapshot
    expect(container.firstChild).toMatchSnapshot();
  });
});
