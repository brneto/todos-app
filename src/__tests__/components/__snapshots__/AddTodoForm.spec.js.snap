// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AddTodo should call handle methods 1`] = `
<header>
  <h1
    class="AddTodoForm__Title-sc-14qkjg5-0 hEqYMt"
  >
    todos
  </h1>
  <input
    class="AddTodoForm__Input-sc-14qkjg5-1 cQsMZo"
    name="todo"
    placeholder="What needs to be done?"
    value=""
  />
</header>
`;
